api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -120,7 +120,7 @@ public static Expression not(Expression child) {
   }
 
   public static <T> UnboundPredicate<T> notNull(UnboundTerm<T> expr) {
-    return new UnboundPredicate<>(Expression.Operation.IS_NULL, expr);
+    return new UnboundPredicate<>(Expression.Operation.NOT_NULL, expr);
```

Comment:
I don't see any uses of this factory method in Spark, Flink, or MR/Hive so luckily, I don't think that this affects engines. (And I would expect it to be caught quickly by Spark tests if we did.) I think that decreases the urgency of this fix, but it still affects API users.

Thanks for catching this, @yyanyy!